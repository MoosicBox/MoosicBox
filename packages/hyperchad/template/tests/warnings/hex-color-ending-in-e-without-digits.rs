//! Test case for hex colors ending in 'e' without digits after
//!
//! This pattern fails at Rust's lexer level because "12e" is interpreted
//! as incomplete scientific notation (expected format: 12e5, 12e10, etc.)
//!
//! Workaround: Use string literal syntax instead: color="#12e"

use hyperchad_template::container;

fn main() {
    // This will fail with: "expected at least one digit in exponent"
    let _ = container! {
        div color=#12e { "This should cause a lexer error" }
    };
}
