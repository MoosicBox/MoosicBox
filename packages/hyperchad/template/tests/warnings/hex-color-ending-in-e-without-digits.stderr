error: expected at least one digit in exponent
  --> tests/warnings/hex-color-ending-in-e-without-digits.rs:13:20
   |
13 |         div color=#12e { "This should cause a lexer error" }
   |                    ^^^
